semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -272,141 +289,102 @@ message Type {
 }
 ```
 
-`Type` represents expression types and signatures of definitions.
-At the moment, types are are modelled after the Scala pickle format [\[26\]][26].
-In the future, we are planning to pay more attention to other languages.
+`Type` represents expression types and definition signatures.
 
-In this section, we describe various alternatives of `Type`, providing example
-SemanticDB data that corresponds to different Scala types,
-inspired by the scala.reflect documentation [\[27\]][27].
-
-In these examples, we will be using a simple notation to describe SemanticDB data.
-In this notation, `M(v1, v2, ...)` corresponds a message `M` with fields set to values
-`v1`, `v2`, etc. Literals correspond to scalar values, and `List(x1, x2, ...)`
-corresponds to repeated values. Moreover, `<X>` corresponds to a message that
-represents `X`.
+The SemanticDB type system is a language-agnostic superset of the type systems
```

Comment:
I'm weary of claiming "language-agnostic" when there is only support for one language. I'm sure there are tons of missing categories of types that not covered here but are used in other mainstream languages.

Reply:
The next line says "of supported languages", which adds up to "The SemanticDB type system is a language-agnostic superset of the type systems of supported languages", so this is not a claim that the SemanticDB type system should be enough for everyone.

What I meant here is that we make a commitment for the type system to be abstracted away from concrete languages, and I think this diff is a nice improvement in this direction. For example, we have renamed compound types and type lambdas (Scala-isms) to the language-agnostic notions of structural types and universal types.

Reply:
No idea how whether this will work actually, but I think we ought to try.

Reply:
I would try at least to dampen down the claims, it's easy to misinterpret "language agnostic" as naive wishful thinking if you are type system researcher. And non-experts might believe this is the holy grail, which it's not either.

Reply:
Sounds good.